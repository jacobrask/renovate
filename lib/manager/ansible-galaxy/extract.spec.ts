import { readFileSync } from 'fs';
import extractPackageFile from './extract';

const yamlFile1 = readFileSync(
  'lib/manager/ansible-galaxy/__fixtures__/requirements01.yml',
  'utf8'
);
const yamlFile2 = readFileSync(
  'lib/manager/ansible-galaxy/__fixtures__/requirements02.yml',
  'utf8'
);
const helmRequirements = readFileSync(
  'lib/manager/ansible-galaxy/__fixtures__/helmRequirements.yml',
  'utf8'
);
const collections1 = readFileSync(
  'lib/manager/ansible-galaxy/__fixtures__/collections1.yml',
  'utf8'
);
const collections2 = readFileSync(
  'lib/manager/ansible-galaxy/__fixtures__/collections2.yml',
  'utf8'
);
const galaxy = readFileSync(
  'lib/manager/ansible-galaxy/__fixtures__/galaxy.yml',
  'utf8'
);

describe('lib/manager/ansible-galaxy/extract', () => {
  describe('extractPackageFile()', () => {
    it('returns null for empty', () => {
      expect(extractPackageFile('nothing here', 'requirements.yml')).toBeNull();
    });
    it('extracts multiple dependencies from requirements.yml', () => {
      const res = extractPackageFile(yamlFile1, 'requirements.yml');
      expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(10);
    });
    it('extracts dependencies from a not beautified requirements file', () => {
      const res = extractPackageFile(yamlFile2, 'requirements.yml');
      expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(2);
    });
    it('check if an empty file returns null', () => {
      const res = extractPackageFile('\n', 'requirements.yml');
      expect(res).toBeNull();
    });
    it('check if a requirements file of other systems returns null', () => {
      const res = extractPackageFile(helmRequirements, 'requirements.yml');
      expect(res).toBeNull();
    });
    it('check collection style requirements file', () => {
      const res = extractPackageFile(collections1, 'requirements.yml');
      expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(13);
      expect(res.deps.filter((value) => value.skipReason != null)).toHaveLength(
        6
      );
    });
    it('check collection style requirements file in reverse order and missing empty line', () => {
      const res = extractPackageFile(collections2, 'requirements.yml');
      expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(4);
    });
    it('check galaxy definition file', () => {
      const res = extractPackageFile(galaxy, 'galaxy.yml');
      expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(2);
    });
  });
});
