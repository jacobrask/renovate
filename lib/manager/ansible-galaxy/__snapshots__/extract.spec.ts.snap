// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/ansible-galaxy/extract extractPackageFile() check collection style requirements file 1`] = `
Array [
  Object {
    "currentValue": "0.9.3",
    "datasource": "galaxy-collection",
    "depName": "geerlingguy.php_roles",
    "depType": "collection",
    "registryUrls": Array [
      "https://galaxy.ansible.com",
    ],
  },
  Object {
    "currentValue": "1.2.2",
    "datasource": "galaxy-collection",
    "depName": "davidban77.gns3",
    "depType": "collection",
    "registryUrls": Array [],
  },
  Object {
    "currentValue": "1.2.2",
    "datasource": "github-tags",
    "depName": "github.com/organization/repo_name",
    "depType": "collection",
    "lookupName": "https://github.com/organization/repo_name.git",
  },
  Object {
    "currentValue": "1.2.2",
    "datasource": "git-tags",
    "depName": "example.com/organization/repo_name",
    "depType": "collection",
    "lookupName": "https://example.com/organization/repo_name.git",
  },
  Object {
    "currentValue": "1.2.2",
    "datasource": "git-tags",
    "depName": "example.com/organization/repo_name",
    "depType": "collection",
    "lookupName": "https://example.com/organization/repo_name.git",
  },
  Object {
    "datasource": "galaxy-collection",
    "depName": "f5networks.f5_modules",
    "depType": "collection",
    "registryUrls": Array [
      "https://cloud.redhat.com/api/automation-hub/",
    ],
    "skipReason": "no-version",
  },
  Object {
    "currentValue": null,
    "datasource": "git-tags",
    "depName": "foo.bar/organization/repo_name",
    "depType": "collection",
    "lookupName": "https://foo.bar/organization/repo_name.git",
    "skipReason": "no-version",
  },
  Object {
    "currentValue": "1.0.0",
    "depName": "fooBar",
    "depType": "collection",
    "skipReason": "no-source-match",
  },
  Object {
    "depName": "foo.Bar",
    "depType": "collection",
    "skipReason": "unsupported",
  },
  Object {
    "depName": "foo.Bar",
    "depType": "collection",
    "skipReason": "local-dependency",
  },
  Object {
    "currentValue": "1.2.2",
    "datasource": "git-tags",
    "depName": "example.com/organization/repo_name",
    "depType": "collection",
    "lookupName": "https://example.com/organization/repo_name.git",
    "skipReason": "unsupported",
  },
  Object {
    "currentValue": "1.9.6",
    "datasource": "galaxy",
    "depName": "geerlingguy.java",
    "depType": "role",
    "lookupName": "geerlingguy.java",
  },
  Object {
    "currentValue": "2.9.0",
    "datasource": "galaxy",
    "depName": "geerlingguy.docker",
    "depType": "role",
    "lookupName": "geerlingguy.docker",
  },
]
`;

exports[`lib/manager/ansible-galaxy/extract extractPackageFile() check collection style requirements file in reverse order and missing empty line 1`] = `
Array [
  Object {
    "currentValue": "0.9.3",
    "datasource": "galaxy-collection",
    "depName": "geerlingguy.php_roles",
    "depType": "collection",
    "registryUrls": Array [
      "https://galaxy.ansible.com",
    ],
  },
  Object {
    "currentValue": "1.2.2",
    "datasource": "galaxy-collection",
    "depName": "davidban77.gns3",
    "depType": "collection",
  },
  Object {
    "currentValue": "1.9.6",
    "datasource": "galaxy",
    "depName": "geerlingguy.java",
    "depType": "role",
    "lookupName": "geerlingguy.java",
  },
  Object {
    "currentValue": "2.9.0",
    "datasource": "galaxy",
    "depName": "geerlingguy.docker",
    "depType": "role",
    "lookupName": "geerlingguy.docker",
  },
]
`;

exports[`lib/manager/ansible-galaxy/extract extractPackageFile() check galaxy definition file 1`] = `
Array [
  Object {
    "currentValue": "1.4.0",
    "datasource": "galaxy-collection",
    "depName": "ansible.windows",
    "depType": "collection",
  },
  Object {
    "currentValue": ">=1.0.0,<2.0.0",
    "datasource": "galaxy-collection",
    "depName": "community.windows",
    "depType": "collection",
  },
]
`;

exports[`lib/manager/ansible-galaxy/extract extractPackageFile() extracts dependencies from a not beautified requirements file 1`] = `
Array [
  Object {
    "currentValue": "0.1.0",
    "datasource": "galaxy",
    "depName": "yatesr.timezone",
    "depType": "role",
    "lookupName": "yatesr.timezone",
  },
  Object {
    "currentValue": "0.1",
    "datasource": "git-tags",
    "depName": "mygroup/ansible-base",
    "depType": "role",
    "lookupName": "git@gitlab.company.com:mygroup/ansible-base.git",
  },
]
`;

exports[`lib/manager/ansible-galaxy/extract extractPackageFile() extracts multiple dependencies from requirements.yml 1`] = `
Array [
  Object {
    "currentValue": "0.1.0",
    "datasource": "galaxy",
    "depName": "yatesr.timezone",
    "depType": "role",
    "lookupName": "yatesr.timezone",
  },
  Object {
    "currentValue": "0.19.0",
    "datasource": "galaxy",
    "depName": "cloudalchemy.node-exporter",
    "depType": "role",
    "lookupName": "cloudalchemy.node-exporter",
  },
  Object {
    "currentValue": "master",
    "datasource": "git-tags",
    "depName": "nginx_role",
    "depType": "role",
    "lookupName": "https://github.com/bennojoy/nginx",
  },
  Object {
    "currentValue": "v1.4",
    "datasource": "git-tags",
    "depName": "willthames/git-ansible-galaxy",
    "depType": "role",
    "lookupName": "http://bitbucket.org/willthames/git-ansible-galaxy",
  },
  Object {
    "currentValue": "0.1",
    "datasource": "git-tags",
    "depName": "mygroup/ansible-base",
    "depType": "role",
    "lookupName": "git@gitlab.company.com:mygroup/ansible-base.git",
  },
  Object {
    "currentValue": "0.1",
    "datasource": "git-tags",
    "depName": "testGroup/testProject",
    "depType": "role",
    "lookupName": "ssh://git@gitlab.company.com/testGroup/testProject.git",
  },
  Object {
    "currentValue": "0.1",
    "datasource": "git-tags",
    "depName": "testGroup/testProject2",
    "depType": "role",
    "lookupName": "ssh://gitlab.company.com/testGroup/testProject2.git",
  },
  Object {
    "currentValue": "3.1",
    "datasource": "git-tags",
    "depName": "testGroup/testProject3",
    "depType": "role",
    "lookupName": "ssh://git@gitlab.company.com:23/testGroup/testProject3.git",
  },
  Object {
    "currentValue": "0.14",
    "datasource": "git-tags",
    "depName": "mygroup/ansible-base",
    "depType": "role",
    "lookupName": "git@gitlab.company.com:mygroup/ansible-base.git",
  },
  Object {
    "currentValue": "0.14",
    "datasource": "galaxy",
    "depName": "company.com:mygroup/ansible-base.git",
    "depType": "role",
    "lookupName": "company.com:mygroup/ansible-base.git",
  },
]
`;
